ciao_contrib/cda/data.py

```diff
@@ -340,42 +341,50 @@ def __init__(self, obsid, base_url, hdr):
         self.header = hdr
 
         ostr = str(obsid)
-        urlname = "{}/{}/{}".format(base_url, ostr[-1], ostr)
-        V3("Looking for directory: {}".format(urlname))
+        urlname = f"{base_url}/{ostr[-1]}/{ostr}"
+        V3(f"Looking for directory: {urlname}")
 
         try:
             urls = downloadutils.find_all_downloadable_files(urlname, hdr)
 
         except urllib.error.HTTPError as herr:
-            V3("HTTPError for {}".format(urlname))
+            V3(f"HTTPError for {urlname}")
             V3(str(herr))
             if herr.code == 404:
-                emsg = "There is no directory {}".format(urlname)
+                emsg = f"There is no directory {urlname}"
             else:
-                emsg = "Unable to access {}\ncode={}".format(urlname,
-                                                             herr.code)
+                emsg = f"Unable to access {urlname}\ncode={herr.code}"
+
             raise IOError(emsg)
 
         except urllib.error.URLError as uerr:
-            V3("URLError for {}".format(urlname))
+            V3(f"URLError for {urlname}")
             V3(str(uerr))
-            emsg = "Unable to reach {}\n{}".format(urlname, uerr.reason)
+            emsg = f"Unable to reach {urlname}\n{uerr.reason}"
             raise IOError(emsg)
 
         self.files = [ObsIdFile(obsid, url) for url in urls]
-        V3("Found {} files".format(len(self.files)))
+        V3(f"Found {len(self.files)} files")
 
     def filter_files(self, types=None, formats=None):
         """Filter the list of files by the given types and/or formats.
+
+        If types is None then we exclude the vvref file type; that is if
+        you want to download the vvref2.pdf file you need to supply
+        a list of types and include the ''vvref' option. This is not
+        a great design!
         """
 
-        V3("Before filtering: {} files".format(len(self.files)))
-        if types is not None:
+        V3(f"Before filtering: {len(self.files)} files")
+        if types is None:
+            # exclude the vvref file
+            self.files = [f for f in self.files if not f.is_type(['vvref'])]
```

Comment:
Humm ... I'm not sure.  Might need others to comment.

If I don't specify any filetypes then I'm either expecting everything or just the most useful files. 

I can also easily argue that the secondary/aspect, secondary/ephem, and oif.fits are not generally useful.  I've never used the ephem/ files and only used the osol files once.  The aspect adat71 files needed for monitor star observations are in primary/ when they it was requested (otherwise they are in 'supporting' only available via chaser).  axbary and sso_freeze use the orbit ephem which is in the primary dir.   These are all small, but often a bunch of them.

Just don't want to start a turf war w/ archive as to what should/shouldn't be in the standard distribution.

If you leave out the condition with `if not f.is_type(['vvref'])` would that keep those files when no filetypes is specified and still let 'vv' just d/l the vv2.pdf file?

Reply:
I believe what the user actually wants is that they just want to just not have the vvref2.pdf file downloaded without them having to make any change. If we want to do this then you have the trickiness of the last commit in some form (and then we can think about excluding  other file types by default).

So options include
 - some form of the current code
 - explicitly come up with a list of file types we want to download by default (perhaps with a label at the download "standard" and get a known set of files)
 - add a way to say "don't download these file types" and then tell the user to set this to vvref (which negates the "user does not need to think about this", but is conceptually cleaner)

The way the current "labelling" code works is that we should have a label for all file types, so I don't understand your last point.

Reply:
I think I'm going to go back to 

1. `download_chandra_obsid obsid` downloads everything (so the helpdesk user will need to jump to point 2)
2. add a `--exclude` flag (name TBD) so you can `download_chandra_obsid --exclude vvref obsid`to exclude the vvfref2 file
3. come up with some decision if you include and exclude a particular file type...

Reply:
:ok:

Note: i'm off today so I won't be able to review anything till Monday.